src/esp32/hcsr04.rs

```diff
@@ -0,0 +1,331 @@
+// Support for HC-SR04 style ultrasonic ranging modules. See
+// https://cdn.sparkfun.com/datasheets/Sensors/Proximity/HCSR04.pdf.
+//
+// Example configuration
+//
+// {
+//   "model": "ultrasonic",
+//   "name": "ultrasonic-sensor",
+//   "type": "sensor",
+//   "attributes": {
+//     "trigger_pin": "15",
+//     "echo_interrupt_pin": "18"
+//     "timeout_ms" : "20",
+//   },
+//   "depends_on": [
+//     "board"
+//   ]
+// }
+//
+// Configuration details:
+//
+// The sensor requires a board dependency in the `depends_on`
```

Comment:
I don't think we actually use the depends_on in micro-RDK. All of our dependencies are implicit and done by registering a [dependency getter](https://github.com/viamrobotics/micro-rdk/blob/6c3b3a5c175fe21ff2c42f258ded231305b7d1ce/src/common/registry.rs#L287)